src/connection-manager/index.js

```diff
@@ -122,21 +172,63 @@ class ConnectionManager {
    * @param {Connection} connection
    */
   onConnect (connection) {
+    if (!Connection.isConnection(connection)) {
+      throw errcode(new Error('conn must be an instance of interface-connection'), ERR_INVALID_PARAMETERS)
+    }
```

Comment:
`onConnect` is an internal handler, we should avoid throwing errors as this is going to be fatal. Log and return instead.